src/BuildOpt.h

```diff
@@ -105,833 +65,218 @@
-    #define RADIOLIB_CB_ARGS_SPI_BEGIN_TRANSACTION      (void, SPIbeginTransaction, void)
-    #define RADIOLIB_CB_ARGS_SPI_TRANSFER               (uint8_t, SPItransfer, uint8_t b)
-    #define RADIOLIB_CB_ARGS_SPI_END_TRANSACTION        (void, SPIendTransaction, void)
-    #define RADIOLIB_CB_ARGS_SPI_END                    (void, SPIend, void)
+  // ... and for the grand finale, we have millis() and micros() DEFINED AS MACROS!
+  #if defined(millis)
+  #undef millis
+  inline unsigned long millis() { return((unsigned long)(STCV / 1000)); };
+  #endif
 
+  #if defined(micros)
+  #undef micros
+  inline unsigned long micros() { return((unsigned long)(STCV)); };
   #endif
+#elif defined(TEENSYDUINO)
+  // Teensy
+  #define RADIOLIB_PLATFORM                           "Teensy"
+#else
+  // other Arduino platforms not covered by the above list - this may or may not work
+  #define RADIOLIB_PLATFORM                           "Unknown Arduino"
+  #define RADIOLIB_UNKNOWN_PLATFORM
 #endif
 
+  #if !defined(RADIOLIB_NC)
+  #define RADIOLIB_NC                                 (0xFF)
+  #endif
+  #if !defined(RADIOLIB_DEFAULT_SPI)
+  #define RADIOLIB_DEFAULT_SPI                        SPI
+  #endif
+  #if !defined(RADIOLIB_DEFAULT_SPI_SETTINGS)
+  #define RADIOLIB_DEFAULT_SPI_SETTINGS               SPISettings(2000000, MSBFIRST, SPI_MODE0)
+  #endif
+  #if !defined(RADIOLIB_NONVOLATILE)
+  #define RADIOLIB_NONVOLATILE                        PROGMEM
+  #endif
+  #if !defined(RADIOLIB_NONVOLATILE_READ_BYTE)
+  #define RADIOLIB_NONVOLATILE_READ_BYTE(addr)        pgm_read_byte(addr)
+  #endif
+  #if !defined(RADIOLIB_NONVOLATILE_READ_DWORD)
+  #define RADIOLIB_NONVOLATILE_READ_DWORD(addr)       pgm_read_dword(addr)
+  #endif
+  #if !defined(RADIOLIB_TYPE_ALIAS)
+  #define RADIOLIB_TYPE_ALIAS(type, alias)            using alias = type;
+  #endif
+
+  #if !defined(RADIOLIB_ARDUINOHAL_PIN_MODE_CAST)
+  #define RADIOLIB_ARDUINOHAL_PIN_MODE_CAST
+  #endif
+  #if !defined(RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST)
+  #define RADIOLIB_ARDUINOHAL_PIN_STATUS_CAST
+  #endif
+  #if !defined(RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST)
+  #define RADIOLIB_ARDUINOHAL_INTERRUPT_MODE_CAST
+  #endif
 #else
   // generic non-Arduino platform
   #define RADIOLIB_PLATFORM                           "Generic"
 
-  // platform properties may be defined here, or somewhere else in the build system
-  #include "noarduino.h"
+  #define RADIOLIB_NC                                 (0xFF)
+  #define RADIOLIB_NONVOLATILE
+  #define RADIOLIB_NONVOLATILE_READ_BYTE(addr)        (*((uint8_t *)(void *)(addr)))
+  #define RADIOLIB_NONVOLATILE_READ_DWORD(addr)       (*((uint32_t *)(void *)(addr)))
+  #define RADIOLIB_TYPE_ALIAS(type, alias)            using alias = type;
+
+  #if !defined(RADIOLIB_DEBUG_PORT)
+    #define RADIOLIB_DEBUG_PORT   stdout
+  #endif
+
+  #define DEC 10
+  #define HEX 16
+  #define OCT 8
+  #define BIN 2
+
+  #include <algorithm>
+  #include <stdint.h>
 
+  typedef uint8_t byte;
```

Comment:
As per the conversation in the PR, this can likely be removed.

Reply:
Yes I forgot to remove it, since there is yet some files using `byte` in the lib.

Reply:
> there is yet some files using byte in the lib.

Where exactly is this?

Reply:
https://github.com/jgromes/RadioLib/blob/88549baf8326ee9f679e09e04f7b3c1a0e8d7a2f/src/protocols/Morse/Morse.h#L127 via gh code search.

Reply:
Ah, I see - that should be `uint8_t`, feel free to change it (or I can do after merging this).